Project.ipynb

```diff
@@ -149,10 +153,64 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def add_temporal_dim(A,n_updates):\n",
+    "## disable some links for a certain amount of time\n",
+    "##\n",
+    "## modes:\n",
+    "## light mode: few links, short time\n",
+    "## heavy mode: few links, long time\n",
+    "## unstable mode: many links, short time\n",
+    "## extreme mode: many links, long time\n",
+    "##\n",
+    "## priority: random, near, far\n",
+    "## prioritize disabling of links involving nodes near or far from Earth\n",
+    "## \n",
+    "## custom_n_targets = number of targets\n",
+    "## custom_offtimes: array containing possible values for off times expressed in number of updates of A\n",
+    "\n",
+    "def disable_links(A,n_updates, mode=\"light\",priority=\"random\", custom_n_targets = None, custom_offtimes = None):\n",
+    "\n",
+    "    arrays = [A for i in range(n_updates)]\n",
+    "    At = np.stack(arrays,axis=2)      #adds temporal dimension\n",
+    "    \n",
+    "    t_nodes = np.arange(n)\n",
+    "    \n",
+    "    if priority == \"random\": np.random.shuffle(t_nodes)   #nodes targeted for link removal\n",
+    "    elif priority == \"near\": t_nodes = sorted_nodes[:int(np.ceil(n/3))]\n",
+    "    elif priority == \"far\": t_nodes = np.flip(sorted_nodes)[:int(np.ceil(n/3))]\n",
```

Comment:
Put instructions on a different line from the if statements.
Don't use variables defined outside the function inside the function, either recalculate the values or pass them as arguments